src/NServiceBus.Core/Transports/Learning/DirectoryBasedTransaction.cs

```diff
@@ -1,21 +1,22 @@
 namespace NServiceBus
 {
-    using System;
     using System.Collections.Generic;
     using System.IO;
     using System.Threading.Tasks;
 
     class DirectoryBasedTransaction : ILearningTransportTransaction
     {
-        public DirectoryBasedTransaction(string basePath, bool immediateDispatch)
+        public DirectoryBasedTransaction(string basePath, string transactionId, bool immediateDispatch)
         {
             this.basePath = basePath;
             this.immediateDispatch = immediateDispatch;
-            var transactionId = Guid.NewGuid().ToString();
 
-            transactionDir = Path.Combine(basePath, ".pending", transactionId);
+            var rootCommitDir = Path.Combine(basePath, CommittedDirName);
 
-            commitDir = Path.Combine(basePath, ".committed", transactionId);
+            Directory.CreateDirectory(rootCommitDir);
```

Comment:
do we need to try to create this directory every time we instantiate a transaction?